pkg/cluster/internal/providers/provider/common/proxy.go

```diff
@@ -60,5 +75,35 @@ func getProxyEnvs(cfg *config.Cluster, getEnv func(string) string) map[string]st
 		envs[NOProxy] = noProxy
 		envs[strings.ToLower(NOProxy)] = noProxy
 	}
-	return envs
+	return envs, nil
+}
+
+// obtains proxy information from docker via docker info against cmder
+func getProxyEnvFromDocker(cmder exec.Cmder) (map[string]string, error) {
+	// get raw fields
+	lines, err := exec.OutputLines(
+		cmder.Command(
+			"docker", "info",
+			"--format",
+			// one per line, upper(key)=value
+			"HTTP_PROXY={{.HTTPProxy}}\nHTTPS_PROXY={{.HTTPSProxy}}\nNO_PROXY={{.NoProxy}}",
+		),
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get proxy information from docker")
+	}
+	// parse out environment settings from the format we specified
+	envs := make(map[string]string)
+	for _, line := range lines {
```

Comment:
we should check that the number of lines is 3 and the names of the variables `[]string{HTTPProxy, HTTPSProxy, NOProxy}`
 to be more robust.
If docker outputs whatever message instead of the variables we are going to return this wrong info

Reply:
- we control the forms
- error output goes to stderr, not stdout